dts/bindings/sensor/nxp,s32-qdec.yaml

```diff
@@ -0,0 +1,96 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Quadrature Decoder driver which processes encoder signals to determine motor revs
+  with the cooperation of S32 IP blocks- eMIOS, TRGMUX and LCU.
+  The sensor qdec application can be used for testing this driver.
+  The following example uses TRGMUX IN2 and IN3 to connect to LCU1 LC0 I0 and I1.
+  LCU1 LC0 O2 and O3 connect to eMIOS0 CH6(Clockwise rotation) and
+  CH7(Counter Clockwise rotation) via TRGMUX_INT_OUT37 and TRGMUX_INT_OUT38
+  micro-ticks-per-rev is set as per vehicle gearbox reduction.
+  lcu output filters are set to capture maximum speed sensitivity and avoid channel noise.
+
+   qdec0 {
+        compatible = "nxp,qdec-s32";
+        pinctrl-0 = <&qdec_s32>;
+        pinctrl-names = "default";
+        micro-ticks-per-rev = <685440000>;
+        status = "okay";
+        trgmux-inst = <0>;
+        trgmux-io-config =
+            <0 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH6 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I2>,
+            <1 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH7 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I3>,
+            <2 TRGMUX_IP_OUTPUT_LCU1_0_INP_I0            TRGMUX_IP_INPUT_SIUL2_IN2>,
+            <3 TRGMUX_IP_OUTPUT_LCU1_0_INP_I1            TRGMUX_IP_INPUT_SIUL2_IN3>;
+        lcu-inst = <1>;
+        lcu-output-filter-config =
+            /* LCU Out HW ID, Rise Filter, Fall Filter */
+            <0 5 5>, /* LCU O0 */
+            <1 5 5>, /* LCU O1 */
+            <2 2 2>, /* LCU O2 */
+            <3 2 2>; /* LCU O3 */
+        emios = <&emios0>;
+        emios-channels = <6 7>;
+   };
+
+compatible: "nxp,qdec-s32"
+
+include: [pinctrl-device.yaml, sensor-device.yaml]
+
+properties:
+  micro-ticks-per-rev:
+    type: int
+    description: |
+      This is a number that is used to determine how many revolutions * 1000000
+      were done based on the current counter's value.
+
+  trgmux-inst:
+    type: int
```

Comment:
This should be a phandle

Reply:
Currently there is no separate driver and DT node for trgmux and lcu in Zephyr, phandle cannot be used.
Working on separate drivers for trgmux and lcu has been added in 
https://github.com/zephyrproject-rtos/zephyr/issues/65459

Reply:
You don't need a driver; devicetree bindings and corresponding compatible nodes would be sufficient to get the base address of the trgmux from devicetree.

Reply:
then you will need to convert the base address to peripheral hw index, because the HAL is index-based. That can be done with the help of some macros.

Reply:
Done !!